classes/class-plugin-upgrade-handler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Handle plugin onboarding and upgrade task related functionality.
+ *
+ * @package Progress_Planner
+ */
+
+namespace Progress_Planner;
+
+/**
+ * Plugin_Upgrade_Handler class.
+ */
+class Plugin_Upgrade_Handler {
+
+	/**
+	 * IDs of the task providers that should be shown to user during onboarding or updating the plugin.
+	 *
+	 * @var array
+	 */
+	private $whitelisted_task_provider_ids;
```

Comment:
Could we change `whitelisted` to `allowed`?
Some people in the community take offense with it